be/src/storage/rowset/beta_rowset.cpp

```diff
@@ -242,7 +241,7 @@ Status BetaRowset::get_segment_iterators(const vectorized::Schema& schema, const
     RETURN_IF_ERROR(load());
 
     vectorized::SegmentReadOptions seg_options;
-    seg_options.block_mgr = options.block_mgr;
+    ASSIGN_OR_RETURN(seg_options.block_mgr, fs::fs_util::block_manager(_rowset_path));
```

Comment:
Get block manager when rowset initializes?

Reply:
Not all rowsets will be visited, create block manager for each may result in unnecessary memory allocation.
But yes, getting a block manager at first may improve the performance or memory usage under heavy query requests.
I'm not sure which one is better.

Reply:
why not pass block_mgr to constructor? `BetaRowset::BetaRowset(const TabletSchema* schema, string rowset_path, RowsetMetaSharedPtr rowset_meta, fs::BlockManager* block_mgr)`

Reply:
An API like this requires callers to know how to create a BlockManager suitable for the path `rowset_path`, this will increase the burden of callers.